messaging/src/main/java/org/axonframework/tracing/NestingSpanFactory.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2010-2022. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.tracing;
+
+import org.axonframework.common.BuilderUtils;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.messaging.Message;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.function.Supplier;
+
+/**
+ * Implementation of {@link SpanFactory} that wraps another factory and creates nested spans when handling messages
```

Comment:
As the nesting behavior is event-specific, I'd expect the JavaDoc like so:

```suggestion
 * Implementation of {@link SpanFactory} that wraps another factory and creates nested spans when handling events
```

Reply:
Updated the text, making it more clear that all messages are nested.

Reply:
Thanks for that, it's a lot clearer now.